apis/actions.md

```diff
@@ -0,0 +1,123 @@
+
+```vala
+var quit_action = new SimpleAction ("quit", null);
+
+add_action (quit_action);
+set_accels_for_action ("app.quit",  {"<Control>q", "<Control>w"});
+```
+
+and this to the end of the `activate ()` method:
+
+```vala
+quit_action.activate.connect (() => {
+    main_window.destroy ();
+});
+```
+
+You'll notice that we do a few things here:
+  * Instantiate a new [`GLib.SimpleAction`](https://valadoc.org/gio-2.0/GLib.SimpleAction.html) with the name "quit"
+  * Add the action to our `Gtk.Application`'s [`ActionMap`](https://valadoc.org/gio-2.0/GLib.ActionMap.html)
+  * Set the "accelerators" (keyboard shortcuts) for "app.quit" to `<Control>q` and `<Control>w"`. Notice that the action name is prefixed with `app`; this refers to the `ActionMap` built in to `Gtk.Application`
+  * Connect to the `activate` signal of our `SimpleAction` and call `destroy ()` on `main_window`. This must be at the end of `activate ()` because of that reference to `main_window`
+
+and now we can tie the action to the HeaderBar Button by assigning the `action_name` property of our Button:
+
+```vala
+var button = new Gtk.Button.from_icon_name ("process-stop", Gtk.IconSize.LARGE_TOOLBAR) {
+    action_name = "app.quit"
+};
+```
+
+Compile your app again and see that you can now quit the app either through the defined keyboard shortcuts or by clicking the Button in the HeaderBar.
+
+{% hint style="info" %}
+Accelerator strings follow a format defined by [`Gtk.accelerator_parse`](https://valadoc.org/gtk+-3.0/Gtk.accelerator_parse.html). You can find a list of key values [on Valadoc](https://valadoc.org/gdk-3.0/Gdk.Key.html)
+{% endhint %}
+
+## Granite.markup_accel_tooltip
+
+There's one more thing we can do here to help improve your app's usability. You may have noticed that in elementary apps you can hover your pointer over tool items to see a description of the button and any available keyboard shortcuts associated with it. We can add the same thing to our Button with [`Granite.markup_accel_tooltip ()`](https://valadoc.org/granite/Granite.markup_accel_tooltip.html).
+
+First, make sure you've included Granite in the build dependencies declared in your meson.build file:
+
+```
+executable(
+    meson.project_name(),
+    'src/Application.vala',
+    dependencies: [
+        dependency('granite'),
+        dependency('gtk+-3.0')
+    ],
+    install: true
+)
+```
+
+Then, set the `tooltip_markup` property of your HeaderBar Button:
+
+```
+var button = new Gtk.Button.from_icon_name ("process-stop", Gtk.IconSize.LARGE_TOOLBAR) {
+    action_name = "app.quit",
+    tooltip_markup = Granite.markup_accel_tooltip (
+        get_accels_for_action ("app.quit"),
+        "Quit"
+    )
+};
+```
+
+It may now be clear why we needed to declare our action at the beginning of `activate ()`: before we can get a list of the accelerators association with the action, we have to define those accelerations and add them to the Application.
```

Comment:
Sp: ".... the accelerators associated with ...."

Also I think the colon should be a hyphen (but not sure).

"It may now be clear why we needed to declare our action at the beginning of `activate ()` - before we can get ...."